src/Servers/Kestrel/Core/src/Internal/Http3/Http3PeerSettings.cs

```diff
@@ -29,6 +33,19 @@ internal List<Http3PeerSetting> GetNonProtocolDefaults()
             list.Add(new Http3PeerSetting(Http3SettingType.MaxFieldSectionSize, MaxRequestHeaderFieldSectionSize));
         }
 
+        if (EnableWebTransport != DefaultEnableWebTransport)
+        {
+            list.Add(new Http3PeerSetting(Http3SettingType.EnableWebTransport, EnableWebTransport));
+
+            // datagrams must be negotiated along with webtransport
+            // https://www.ietf.org/archive/id/draft-ietf-webtrans-http3-01.html#name-establishing-a-transport-ca
+            list.Add(new Http3PeerSetting(Http3SettingType.H3Datagram, H3Datagram));
+        }
+        else if (H3Datagram != DefaultH3Datagram)
+        {
+            list.Add(new Http3PeerSetting(Http3SettingType.H3Datagram, H3Datagram));
```

Comment:
Should this `else-if` even be an option? It is mentioned in the spec, so I added the option, but I don't (currently) intend to create an API to only send individual datagrams without WebTransport.

Reply:
Upon talking to @halter73 Kestrel most likely doesn't even support this yet and, from the sounds of it, this by itself is a whole intern project. So, I am going to remove it. At the very least it is hardcoded to be off right now.